### ISSUE TITLE
[Bug]: Can't set up write-good linter for markdown

### ISSUE THREAD

**Issue by qvieth** [LunarVim/LunarVim#1249]
There is no `lvim.lang.markdown` interface to set up `write-good` linter (null-ls does have support for this linter).

https://github.com/jose-elias-alvarez/null-ls.nvim/blob/main/doc/BUILTINS.md#write-good

**Comment by rebuilt** [comment]
We are looking into this.  In the meantime you can add `!write-good %` to a keybind or if you want to see suggestions on save, to an autocommand

```
lvim.autocommands.custom_groups = {
	{ "BufWritePre", "*.md", "!write-good %" },
}

```

**Comment by kylo252** [comment]
> There is no `lvim.lang.markdown` interface to set up `write-good` linter (null-ls does have support for this linter).

The easiest way for now is just add these lines to your `lvim/config.lua`

```lua
local null_ls = require "null-ls"
null_ls.register { sources = { null_ls.builtins.diagnostics.write_good } }
```

You can also create a file `~/.config/lvim/ftplugin/markdown.lua` and them there.

---

@Tastyep, do you think it's possible to fix the dash conversion issue? 

```lua
 lvim.lang.markdown = {
  linters = { { exe = "write-good" } },
  formatters = { { exe = "prettier" } },
}
require("lsp.null-ls").setup "markdown"
```

will result in `Not a valid linter: write-good` because it should be evaluating `write_good` instead.

**Comment by Tastyep** [comment]
> @Tastyep, do you think it's possible to fix the dash conversion issue?
> 
> ```lua
>  lvim.lang.markdown = {
>   linters = { { exe = "write-good" } },
>   formatters = { { exe = "prettier" } },
> }
> require("lsp.null-ls").setup "markdown"
> ```
> 
> will result in `Not a valid linter: write-good` because it should be evaluating `write_good` instead.

https://github.com/LunarVim/LunarVim/pull/1447/commits/022a1a21312477acc720e2129212fc0eab2945c9

**Comment by kylo252** [comment]
Fixed in #1584
